docs/protocol/src/concepts/validators.md

```diff
@@ -3,44 +3,52 @@
 
-The chain chooses a validator limit N as a consensus parameter. When a validator's delegation pool (a) has a nonzero balance and (b) its (voting-power-adjusted) size is in the top N validators, it moves into the active state during the next epoch transition.  Active validators participate in consensus, and are communicated to Tendermint. Stake contributed to an active validator's delegation pool earns rewards (the validator's rates are updated at each epoch to track the rewards accruing to the pool). That stake is bonded, so undelegations have an unbonding period and an output quarantine. An active validator can exit the consensus set in two ways.
+The chain chooses a validator limit N as a consensus parameter. When a validator's delegation pool (a) has a nonzero balance and (b) its (voting-power-adjusted) size is in the top N validators, it moves into the active state during the next epoch transition.  Active validators participate in consensus, and are communicated to Tendermint. Stake contributed to an active validator's delegation pool earns rewards (the validator's rates are updated at each epoch to track the rewards accruing to the pool). That stake is bonded, so undelegations have an unbonding period and an output quarantine. An active validator can exit the consensus set in four ways.
 
-First, the validator could be slashed.  This can happen in any block, not just at an epoch transition.  Slashed validators are immediately removed from the consensus set. Any pending undelegations from a slashed validator are cancelled: the quarantined output notes are deleted, and the quarantined nullifiers are removed from the nullifier set.  The validator's rates are updated to price in the slashing penalty, and are then held constant. Slashed validators are jailed, and permanently prohibited from participation in consensus (though their operators can create new identity keys, if they'd like to). Stake cannot be delegated to a slashed validator. Stake already contributed to a slashed validator's delegation pool is not bonded (the validator has already been slashed and jailed), so undelegations are effective immediately, with no unbonding period and no quarantine.
+First, the validator could be jailed and slashed for inactivity.  This can happen in any block, not just at an epoch transition.  Jailed validators are immediately removed from the consensus set. The validator's rates are updated to price in the slashing penalty, and are then held constant. Validators jailed for inactivity are not permanently prohibited from participation in consensus, and their operators can re-activate them by re-uploading the validator definition. Stake cannot be delegated to a slashed validator. Stake already contributed to a slashed validator's delegation pool will enter an unbonding period to hold the validator accountable for any byzantine behavior during the unbonding period. Re-delegations may occur after the validator enters the "Inactive" state.
```

Comment:
> This can happen in any block, not just at an epoch transition.

This isn't accurate any more: validator changes always happen at epoch transitions, but those transitions are dynamically triggered:

> This can happen in any block, triggering an unscheduled epoch transition.

Reply:
and ditto in the tombstoning section

Reply:
actually maybe we should update the part of the spect that describes epochs to say that they're time based and dynamic?

Reply:
I don't think the docs ever specify the epoch in terms of block height _or_ time, but maybe it should be more clear? Closest I can find is in the `batching_flows.md`:

```markdown
To allow batching across block boundaries,
Penumbra organizes blocks into epochs, and applies changes to the validator set
only at epoch boundaries.  Decryption keys live for the duration of the epoch,
allowing value flows to be batched over any time interval from 1 block up to the
length of an epoch. We propose epoch boundaries on the order of 1-3 days.
```